ci/scripts/github_pr_comment.py

```diff
@@ -0,0 +1,141 @@
+      repository(owner: $owner, name: $name) {
+        pullRequest(number: $number) {
+          title
+          body
+          state
+          number
+          author {
+            login
+          }
+          labels(first:100) {
+            nodes {
+              name
+            }
+          }
+          comments(last: 100) {
+            pageInfo {
+              hasPreviousPage
+            }
+            nodes {
+              author {
+                login
+              }
+              databaseId
+              body
+            }
+          }
+          commits(last: 1) {
+            nodes {
+              commit {
+                oid
+                statusCheckRollup {
+                  contexts(first: 100) {
+                    pageInfo {
+                      hasNextPage
+                    }
+                    nodes {
+                      ... on StatusContext {
+                        state
+                        context
+                        targetUrl
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+"""
+
+
+if __name__ == "__main__":
+    help = "Comment a welcome message on PRs"
+    parser = argparse.ArgumentParser(description=help)
+    parser.add_argument("--remote", default="origin", help="ssh remote to parse")
+    parser.add_argument("--pr", required=True)
+    parser.add_argument("--test-data", help="(testing) mock GitHub API data")
+    parser.add_argument("--test-comments", help="(testing) testing comments")
+    parser.add_argument(
+        "--dry-run",
+        action="store_true",
+        default=False,
+        help="run but don't send any request to GitHub",
+    )
+    args = parser.parse_args()
+    init_log()
+
+    remote = git(["config", "--get", f"remote.{args.remote}.url"])
+    user, repo = parse_remote(remote)
+
+    test_data = None
+    if args.test_data is not None:
+        test_data = json.loads(args.test_data)
+
+    github = GitHubRepo(
+        user=user,
+        repo=repo,
+        token=DRY_RUN if args.dry_run else os.environ["GITHUB_TOKEN"],
+        test_data=test_data,
+    )
+
+    pr_data = github.graphql(
+        PR_QUERY,
+        {
+            "owner": user,
+            "name": repo,
+            "number": int(args.pr),
+        },
+    )
+
+    pr_data = pr_data["data"]["repository"]["pullRequest"]
+    commenter = BotCommentBuilder(github=github, data=pr_data)
+
+    if args.test_comments is not None:
+        test_comments = json.loads(args.test_comments)
+        skipped_tests = test_comments["skipped-tests"]
+        ccs = test_comments["ccs"]
+        docs_info = test_comments["docs"]
+    else:
+        skipped_tests = get_skipped_tests_comment(pr_data, github=github)
+        ccs = get_tags(pr_data, github, team_issue=10317)
+        docs_info = get_doc_url(pr_data)
+
+    items = {
+        "ccs": ccs,
```

Comment:
do we need to also keep any sorta central registry of these? could conceptually put these keys as an enum in/next to BotCommentBuilder

Reply:
this is kind of the central registry already, nothing else outside this script really needs to know the specifics of what the sections are since its all generic